### PR TITLE
feat(sdk): detail option to kfp run get. Fixes #6315

### LINKED ISSUE #6315: [feature] kfp CLI support 'kfp run get <run_id>' in detailed information
### Feature Area

/area sdk

### What feature would you like to see?
Currently, `kfp run get <run_id>` only shows ["run id", "name", "status", "created at"].
However, user could want to see other information of the run.

### What is the use case or pain point?
For my case, I want to fetch my **run's output metrics** with kfp CLI

### Is there a workaround currently?

Since `kfp.Client()` already have `get_run()` method in sdk, which have all information including metrics of run,
I think the UX is only what matters: Which information would be useful or useless, and how user could select them.

For example, I guess `pipeline_Spec.workflow_manifest` could always be omitted, since nobody wants such information.

Therefore, I think with `click.option('--detail')` in get() method in `pipelines/sdk/python/kfp/cli/run.py` shows all information except `pipeline_Spec.workflow_manifest` in json format would be fine.

However, if there is other suggestions, please let me know.
Also, If you guys are ok, I'd like to took this issue and implement this feature.

---

Love this idea? Give it a 👍. We prioritize fulfilling features with the most 👍.

### PR DESCRIPTION
**Pull request by anencore94**
**Description of your changes:**
Fixes #6315
- to get more detailed information when using kfp run get CLI

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

**Screenshot**
- for example
kfp-run-get-detail

### CODE REVIEW

File: sdk/python/kfp/cli/run.py [hunk 1]
```diff
@@ -140,3 +146,13 @@ def _print_runs(runs, output_format):
     headers = ['run id', 'name', 'status', 'created at']
     data = [[run.id, run.name, run.status, run.created_at.isoformat()] for run in runs]
     print_output(data, headers, output_format, table_format='grid')
+
+
+def _print_run_in_detail(client, run_id):
```
Comment (chensun) [inline sdk/python/kfp/cli/run.py:151]: Maybe this can be merged into `_print_runs`?

Comment (anencore94) [reply sdk/python/kfp/cli/run.py:151]: maybe this function could be merged into `_display_run`
`_print_runs` and `print_run_in_detail` are not that similar I think
WDYT ?

Comment (anencore94) [reply sdk/python/kfp/cli/run.py:151]: Maybe something like this:
(Do not support watch with detail)

```
def _display_run(client, namespace, run_id, watch, output_format, detail=False):
    run = client.get_run(run_id).run

    if detail:
        data = {
            key: value.isoformat() if isinstance(value, datetime.datetime) else value
            for key, value in run.to_dict().items()
            if key not in ['pipeline_spec']  # useless but too much detailed field
        }
        click.echo(data)
        return

    _print_runs([run], output_format)
    if not watch:
        return
    argo_path = shutil.which('argo')
...
...

```

Comment (anencore94) [reply sdk/python/kfp/cli/run.py]: Could you re-check ? @chensun Thanks in advance!

**Pull request merged**